src/irmin-pack/unix/pack_store.ml

```diff
@@ -92,9 +92,14 @@ struct
 
   let v ~config ~fm ~dict ~dispatcher =
     let indexing_strategy = Conf.indexing_strategy config in
-    let lru_size = Conf.lru_size config in
+    (* assume the default object size is the size of a hash *)
+    let obj_size = Hash.hash_size in
+    let lru_size = Conf.lru_size config * obj_size in
     let staging = Tbl.create 127 in
-    let lru = Lru.create lru_size in
+    let weight v =
+      match size_of_value v with None -> obj_size | Some n -> n
+    in
```

Comment:
Depending on `Val.t`, the runtime of `size_of` could be prohibitive. 

I think we should use a weight of 1 for everything except for blobs when `contents_length_header = Some 'Varint`.

For blobs, and if `contents_length_header = None`, the user might be using a complex type.

For inodes, I expect `size_of` to be time consuming. Also, inodes may reference other inodes, which would not be reflected in `size_of`.

Reply:
It's probably worth encapsulating this behavior in a `Val.weight` function.